src/java/org/apache/cassandra/sensors/RequestSensors.java

```diff
@@ -74,12 +78,33 @@ public Set<Sensor> getSensors(Type type)
 
     public void incrementSensor(Context context, Type type, double value)
     {
-        Optional.ofNullable(sensors.get(Pair.create(context, type))).ifPresent(s -> s.increment(value));
+        updateLock.readLock().lock();
+        try
+        {
+            Optional.ofNullable(sensors.get(Pair.create(context, type))).ifPresent(s -> s.increment(value));
+        }
+        finally
+        {
+            updateLock.readLock().unlock();
+        }
     }
 
     public void syncAllSensors()
     {
-        sensors.values().forEach(s -> sensorsRegistry.get().updateSensor(s.getContext(), s.getType(), s.getValue()));
+        updateLock.writeLock().lock();
+        try
+        {
+            sensors.values().forEach(sensor -> {
+                double current = latestSyncedValuePerSensor.getOrDefault(sensor, 0d);
+                double update = sensor.getValue() - current;
```

Comment:
Shall we return fast if the `update` is zero?

Reply:
Sure.

Reply:
See: https://github.com/datastax/cassandra/pull/1119/commits/9de62fcc76979eef6cee0a58b9f21ce4f6019cfa